iota-client/src/client.rs

```diff
@@ -134,42 +137,45 @@ impl Client {
 
     /// Sync the node lists per node_sync_interval milliseconds
     pub(crate) fn start_sync_process(
-        mut runtime: Runtime,
+        runtime: &Runtime,
         sync: Arc<RwLock<HashSet<Url>>>,
         nodes: Vec<Url>,
         node_sync_interval: NonZeroU64,
-        stop_sync: Arc<AtomicBool>,
-    ) -> thread::JoinHandle<()> {
-        let node_sync_interval = Duration::from_millis(node_sync_interval.into());
-
-        thread::spawn(move || {
-            while !stop_sync.load(Ordering::Relaxed) {
-                // sleep first since the first `sync_nodes` call is made by the builder
-                // to ensure the node list is filled before the client is used
-                thread::sleep(node_sync_interval);
-                Client::sync_nodes(&mut runtime, &sync, &nodes);
-            }
-        })
+        mut kill: Receiver<()>,
+    ) {
+        let node_sync_interval = TokioDuration::from_millis(node_sync_interval.into());
+
+        runtime.enter(|| {
+            tokio::spawn(async move {
+                tokio::select! {
+                    _ = async {
+                        loop {
+                            // delay first since the first `sync_nodes` call is made by the builder
+                            // to ensure the node list is filled before the client is used
+                            delay_for(node_sync_interval).await;
+                            Client::sync_nodes(&sync, &nodes).await;
+                        }
+                    } => {}
+                    _ = kill.recv() => {}
```

Comment:
Have you had the chance to try this ? I am having a hard time figuring out if this is correct or not

Reply:
I would have looped over a select but this may be right as well, a bit unusual though

Reply:
yeah that was my first try but it wasn't compiling :P but now I got it to work